.github/workflows/fossa.yaml

```diff
@@ -0,0 +1,35 @@
+name: FOSSA
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  fossa:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
```

Comment:
can we used a pinned version?

Reply:
Good call! Updated with those pinned versions